packages/wrangler/templates/startDevWorker/ProxyWorker.ts

```diff
@@ -304,8 +304,23 @@ function insertLiveReloadScript(
  * so that this proxy is transparent to the Client Browser and User Worker.
  */
 function rewriteUrlRelatedHeaders(headers: Headers, from: URL, to: URL) {
+	const setCookie = headers.getAll("Set-Cookie");
+	headers.delete("Set-Cookie");
+	for (const cookie of setCookie) {
+		headers.append(
+			"Set-Cookie",
+			cookie.replace(
+				new RegExp(`Domain=${from.hostname}($|;|,)`),
+				`Domain=${to.hostname}$1`
+			)
+		);
+	}
 	headers.forEach((value, key) => {
-		if (typeof value === "string" && value.includes(from.host)) {
+		if (
+			typeof value === "string" &&
+			value.includes(from.host) &&
+			key !== "set-cookie"
```

Comment:
I think you need to lowercase `key` here since it wouldn't match `Set-Cookie`.

Reply:
Headers are case-insensitive, and judging by the docs `headers.forEach()` will lowercase header name: https://developer.mozilla.org/en-US/docs/Web/API/Headers/forEach